### PR TITLE
Fix/explicit outputs

### PR DESCRIPTION
**Pull request by dnitsch**
Adds Artifacts key to the task 

Adds container native support context

### CODE REVIEW

File: internal/config/context.go [hunk 1]
```diff
@@ -1,54 +1,126 @@
...
+	if defEnvFile == nil && !isContainerContext {
+		return defEnvFile, nil
+	}
+
+	envFile := utils.NewEnvFile(func(e *utils.Envfile) {
+		e.Generate = defEnvFile.Generate
+		if isContainerContext {
+			e.Generate = true
+		}
+		e.Delay = defEnvFile.Delay
```
Comment (ElvenSpellmaker) [inline internal/config/context.go:68]: We can remove the delay feature, it never fixed what it was meant to

File: internal/config/task.go [hunk 2]
```diff
@@ -16,38 +13,36 @@ func buildTask(def *TaskDefinition, lc *loaderContext) (*task.Task, error) {
...
-		}
+	// 	envs, err := utils.ReadEnvFile(filename)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
 
-		t.Env = variables.FromMap(envs).Merge(t.Env)
-	}
+	// 	t.Env = t.Env.Merge(variables.FromMap(envs))
+	// }
```
Comment (ElvenSpellmaker) [inline internal/config/task.go:45]: Commented out?

File: internal/utils/binary.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+// 	return b.baseArgs
+// }
+
+// func (b *Binary) ShellArgs() []string {
+// 	return b.shellArgs
+// }
+
+// func (b *Binary) ContainerArgs() []string {
+// 	return b.containerArgs
+// }
```
Comment (ElvenSpellmaker) [inline internal/utils/binary.go:50]: commented out?

File: pkg/runner/runner_test.go [hunk 4]
```diff
@@ -1,41 +1,41 @@
...
-	runner.SetVariables(variables.FromMap(map[string]string{"Root": "/tmp"}))
-	runner.WithVariable("Root", "/")
+	rnr.Stdout, rnr.Stderr = &bytes.Buffer{}, &bytes.Buffer{}
+	rnr.SetVariables(variables.FromMap(map[string]string{"Root": "/tmp"}))
+	rnr.WithVariable("Root", "/")
 
 	task1 := taskpkg.NewTask("t1")
 	task1.Context = "local"
-	task1.ExportAs = "EXPORT_NAME"
+	// task1.ExportAs = "EXPORT_NAME"
```
Comment (ElvenSpellmaker) [inline pkg/runner/runner_test.go:38]: commented out?

File: pkg/runner/runner_test.go [hunk 5]
```diff
@@ -118,27 +118,27 @@ func Test_DockerExec_Cmd(t *testing.T) {
...
-			runner.SetVariables(variables.FromMap(map[string]string{"Root": "/tmp"}))
-			runner.WithVariable("Root", "/")
+			rnr.Stdout, rnr.Stderr = testOut, testErr
+			rnr.SetVariables(variables.FromMap(map[string]string{"Root": "/tmp"}))
+			rnr.WithVariable("Root", "/")
 
 			task1 := taskpkg.NewTask("default:docker")
 			task1.Context = "default_docker"
-			task1.ExportAs = "EXPORT_NAME"
+			// task1.ExportAs = "EXPORT_NAME"
```
Comment (ElvenSpellmaker) [inline pkg/runner/runner_test.go:134]: commented out?

File: internal/config/context.go [hunk 6]
```diff
@@ -1,54 +1,126 @@
...
+		// if shell is overwritten it should also contain the
+		shellArgs := []string{"sh", "-c"}
+		if def.Container.Shell != "" {
+			// SHELL ARGS
+			shellArgs = []string{def.Container.Shell}
+			if def.Container.ShellArgs != nil {
+				shellArgs = append(shellArgs, def.Container.ShellArgs...)
+			} else {
+				// user should know that this might not work
+				logrus.Warnf("your chosen shell: %s does not include any arguments, usually at least -c as the command gets parsed as string", def.Container.Shell)
```
Comment (ElvenSpellmaker) [inline internal/config/context.go:119]: Do any shells not pass args? 🤔

Comment (dnitsch) [reply internal/config/context.go:119]: well it's more that if you over write the default `sh` with let's say `bash` and do not include shell args like `-c` the command might fail if for example it is quoted with `'`

File: internal/config/context.go (hunk 1 above)
Comment (dnitsch) [reply internal/config/context.go:68]: :D I made this note in the code 
image

File: internal/config/context.go (hunk 6 above)
Comment (ElvenSpellmaker) [reply internal/config/context.go:119]: Yeah that makes sense

**Pull request merged**